app/src/keymap.c

```diff
@@ -247,12 +257,27 @@ int zmk_keymap_set_layer_binding_at_idx(zmk_keymap_layer_id_t layer_id, uint8_t
 
     ASSERT_LAYER_VAL(layer_id, -EINVAL)
 
+    uint32_t *pos_map;
+    int ret = zmk_physical_layouts_get_selected_to_stock_position_map(&pos_map);
+    if (ret < 0) {
+        LOG_WRN("Failed to get the mapping to determine where to set the binding (%d)", ret);
+        return ret;
+    }
+
+    uint32_t storage_binding_idx = pos_map[binding_idx];
```

Comment:
`ret` here is now the size of `pos_map`. Do we need to check that `binding_index` is in bounds?

Also, do we need to check that `storage_binding_index` is in bounds for `zmk_keymap`?

(If the answer to these is "yes, but it's an error in the code rather than a user configuration error if they're out of bounds", then I would probably just use asserts rather than returning an error.)

Reply:
Added checks. Thanks.